Added feature to upload dashboard JSON file to update

LeoMorgan113 opened the pull request:
## Pull Request description

Added functionality to upload a JSON of the dashboard to update the dashboard content.
Before:
<img width="1845" height="923" alt="2025-09-01_13-59_1" src="https://github.com/user-attachments/assets/a6203fae-e05d-4780-af3d-6c8ad5401558" />
After:
<img width="1845" height="921" alt="2025-09-01_13-59" src="https://github.com/user-attachments/assets/e3ea2429-4b69-4b93-bab7-9b886d3cf71c" />

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.
- [ ] If new yml property was added: make sure a description is added (above or near the property).

vvlladd28 commented:
It might be better to add this functionality here. It will be more understandable for the end user and will result in cleaner code.
<img width="1088" height="236" alt="image" src="https://github.com/user-attachments/assets/14c86e26-c263-43a3-b204-99e44c7fdafd" />

LeoMorgan113 commented:
Updated logic, added dialog window for file upload
<img width="1849" height="930" alt="2025-09-02_10-35" src="https://github.com/user-attachments/assets/f4b428ce-0058-413e-8cc5-6c1e7f352989" />
<img width="1853" height="930" alt="2025-09-02_10-35_1" src="https://github.com/user-attachments/assets/365851fd-3c1d-468e-8704-10baaf6e4a26" />

vvlladd28 commented on ui-ngx/src/app/shared/models/dashboard.models.ts:
```diff
@@ -195,6 +195,7 @@ export interface Dashboard extends DashboardInfo {
   configuration?: DashboardConfiguration;
   dialogRef?: MatDialogRef<any>;
   resources?: Array<any>;
+  fileContent?: DashboardConfiguration;
```
Please clear code

vvlladd28 commented on ui-ngx/src/app/modules/home/pages/dashboard/import-dashboard-file-dialog.component.ts:
```diff
@@ -0,0 +1,91 @@
+/// limitations under the License.
+///
+
+import {Component, Inject, OnInit} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material/dialog';
+import {Store} from '@ngrx/store';
+import {AppState} from '@core/core.state';
+import {UntypedFormBuilder, UntypedFormGroup} from '@angular/forms';
+import {DashboardService} from '@core/http/dashboard.service';
+import {Dashboard, DashboardInfo} from '@app/shared/models/dashboard.models';
+import {ActionNotificationShow} from '@core/notification/notification.actions';
+import {TranslateService} from '@ngx-translate/core';
+import {DialogComponent} from '@shared/components/dialog.component';
+import {Router} from '@angular/router';
```
Need format import

vvlladd28 commented on ui-ngx/src/app/modules/home/pages/dashboard/import-dashboard-file-dialog.component.ts:
```diff
@@ -0,0 +1,91 @@
+
+  dashboard: Dashboard;
+  currentFileName: string = '';
+  uploadFileFormGroup: UntypedFormGroup;
```
```suggestion
  uploadFileFormGroup: FormGroup;
```

vvlladd28 commented on ui-ngx/src/app/modules/home/pages/dashboard/import-dashboard-file-dialog.component.ts:
```diff
@@ -0,0 +1,91 @@
+              public translate: TranslateService,
+              private dashboardService: DashboardService,
+              public dialogRef: MatDialogRef<ImportDashboardFileDialogComponent>,
+              public fb: UntypedFormBuilder) {
```
```suggestion
              private fb: FormBuilder) {
```

vvlladd28 commented on ui-ngx/src/app/modules/home/pages/dashboard/import-dashboard-file-dialog.component.ts:
```diff
@@ -0,0 +1,91 @@
+})
+export class ImportDashboardFileDialogComponent extends DialogComponent<ImportDashboardFileDialogComponent> implements OnInit {
+
+  dashboard: Dashboard;
```
```suggestion
  private dashboard: Dashboard;
```

vvlladd28 commented on ui-ngx/src/app/modules/home/pages/dashboard/import-dashboard-file-dialog.component.ts:
```diff
@@ -0,0 +1,91 @@
+  constructor(protected store: Store<AppState>,
+              protected router: Router,
+              @Inject(MAT_DIALOG_DATA) public data: DashboardInfoDialogData,
+              public translate: TranslateService,
```
Not used, need to remove

vvlladd28 commented on ui-ngx/src/app/modules/home/pages/dashboard/import-dashboard-file-dialog.component.ts:
```diff
@@ -0,0 +1,91 @@
+              @Inject(MAT_DIALOG_DATA) public data: DashboardInfoDialogData,
+              public translate: TranslateService,
+              private dashboardService: DashboardService,
+              public dialogRef: MatDialogRef<ImportDashboardFileDialogComponent>,
```
```suggestion
              protected dialogRef: MatDialogRef<ImportDashboardFileDialogComponent>,
```
